include/mbedtls/ssl.h

```diff
@@ -1297,6 +1297,9 @@ struct mbedtls_ssl_session {
 
 #if defined(MBEDTLS_SSL_EARLY_DATA)
     uint32_t MBEDTLS_PRIVATE(max_early_data_size);          /*!< maximum amount of early data in tickets */
+#if defined(MBEDTLS_SSL_ALPN) && defined(MBEDTLS_SSL_SRV_C)
+    char *alpn;                      /*!< ALPN negotiated in the session */
```

Comment:
```suggestion
    char *ticket_alpn;                      /*!< ALPN negotiated in the session where the ticket was generated  */
```
This is an attribute of a ticket thus I prefer to name it `ticket_alpn`. Note that when resuming (not when creating tickets) this may not be the negotiated ALPN for the new session based on the ticket. This is allowed (different ALPN) as long as no early data is involved.
And please move it with the other ticket fields: between `hostname` and `ticket_reception_time`.